pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LinuxBrokerHostUsageImpl.java

```diff
@@ -84,11 +87,12 @@ public LinuxBrokerHostUsageImpl(int hostUsageCheckIntervalMin,
             log.warn("Failed to check cgroup CPU usage file: {}", e.getMessage());
         }
         this.isCGroupsEnabled = isCGroupsEnabled;
+        this.executorService = executorService;
+        this.hostUsageCheckIntervalMin = hostUsageCheckIntervalMin;
 
         // Call now to initialize values before the constructor returns
         calculateBrokerHostUsage();
-        executorService.scheduleAtFixedRate(catchingAndLoggingThrowables(this::calculateBrokerHostUsage),
-                hostUsageCheckIntervalMin,
+        this.executorService.schedule(catchingAndLoggingThrowables(this::calculateBrokerHostUsage),
```

Comment:
I also wonder why we use `scheduleAtFixedRate` instead of `scheduleAtFixedDelay`. When we have long GC, it's not helpful to run `calculateBrokerHostUsage()` many times subsequently in order to keep a constant rate. @lhotari and @merlimat - what do you think?

Reply:
IMHO, `scheduleAtFixedDelay` is better.

Reply:
I agree, using scheduleAtFixedDelay is better

Reply:
I think it is correct to use scheduleAtFixedDelay instead of scheduleAtFixedRate. Otherwise there will be Infinity problems。